site/src/components/Popover/Popover.tsx

```diff
@@ -0,0 +1,163 @@
+import {
+  ReactElement,
+  ReactNode,
+  cloneElement,
+  createContext,
+  useContext,
+  useRef,
+  useState,
+} from "react";
+// This is used as base for the main Popover component
+// eslint-disable-next-line no-restricted-imports -- Read above
+import MuiPopover, {
+  type PopoverProps as MuiPopoverProps,
+} from "@mui/material/Popover";
+
+type TriggerMode = "hover" | "click";
+
+type TriggerRef = React.RefObject<HTMLElement>;
+
+type TriggerElement = ReactElement<{
+  onClick?: () => void;
+  ref: TriggerRef;
+}>;
+
+type PopoverContextValue = {
+  open: boolean;
+  setOpen: React.Dispatch<React.SetStateAction<boolean>>;
+  triggerRef: TriggerRef;
+  mode: TriggerMode;
+};
+
+const PopoverContext = createContext<PopoverContextValue | undefined>(
+  undefined,
+);
+
+export const Popover = (props: {
+  children: ReactNode | ((popover: PopoverContextValue) => ReactNode); // Allows inline usage
```

Comment:
Do you feel like this should support arrays, too?

Reply:
I'm quite sure `ReactNode` does support it already but if not, I think we can add it when we need it.